cassandra-diagnostics-commons/src/main/java/io/smartcat/cassandra/diagnostics/Query.java

```diff
@@ -4,69 +4,112 @@
+    }
 
     /**
      * CQL statement type.
+     * @return CQL statement type
      */
-    public String statementType;
+    public String getStatementType() {
+        return statementType;
+    }
 
     /**
      * Query's key-space name.
+     * @return key space name
      */
-    public String keyspace;
+    public String getKeyspace() {
+        return keyspace;
+    }
 
     /**
      * Query's table name (if applicable).
+     * @return table name
      */
-    public String tableName;
+    public String getTableName() {
+        return tableName;
+    }
 
     /**
      * CQL statement.
+     * @return CQL statement
      */
-    public String statement;
-
-    /**
-     * Default constructor.
-     */
-    public Query() {
+    public String getStatement() {
+        return statement;
     }
 
-    /**
-     * Constructor.
-     *
-     * @param startTimeInMilliseconds     query execution start time
-     * @param executionTimeInMilliseconds query execution time
-     * @param clientAddress               client address
-     * @param statement                   query statement
-     */
-    public Query(long startTimeInMilliseconds, long executionTimeInMilliseconds, String clientAddress,
-            String statement) {
-        super();
+    private Query(long startTimeInMilliseconds, long executionTimeInMilliseconds, String clientAddress,
+            String statementType, String keyspace, String tableName, String statement, String executionErrorMessage) {
         this.startTimeInMilliseconds = startTimeInMilliseconds;
         this.executionTimeInMilliseconds = executionTimeInMilliseconds;
+        this.executionErrorMessage = executionErrorMessage;
         this.clientAddress = clientAddress;
+        this.statementType = statementType;
+        this.keyspace = keyspace;
+        this.tableName = tableName;
         this.statement = statement;
     }
 
+    /**
+     * Returns a new instance of Query.
+     *
+     * @param startTimeInMilliseconds query execution's start time, given as epoch timestamp in milliseconds
+     * @param executionTimeInMilliseconds query execution time in milliseconds
+     * @param clientAddress query's client socket address
+     * @param statementType type of query's statement
+     * @param keyspace query's key space
+     * @param tableName query's table name
+     * @param statement query's CQL statement
+     * @param executionErrorMessage query's execution error message
+     * @return a new Query instance
+     */
+    public static Query create(final long startTimeInMilliseconds, final long executionTimeInMilliseconds,
+            final String clientAddress, final String statementType, final String keyspace, final String tableName,
+            final String statement, final String executionErrorMessage) {
+        return new Query(startTimeInMilliseconds, executionTimeInMilliseconds, executionErrorMessage,
+            clientAddress, statementType, keyspace, tableName, statement);
+    }
```

Comment:
How the builder pattern goes with the idea to have immutable objects that can be created only as fully initialized objects? I guess it would trigger creation of not-that-simple builder methods to ensure the mentioned requirement.